### PR TITLE
✨Added Omni-search

### LINKED ISSUE #13: Omni-search for events table

### LINKED ISSUE #14: Omni-search for contact table

### PR DESCRIPTION
**Pull request by Junnyyy**
What type of PR is this? 🚧
- [x] ✨ Feature

Describe your PR request 🔥
This pull requests adds general omni-search and specific endpoints for /search/events and /search/contacts.

Endpoints can be tested using the following:
http://localhost:5050/search/events?event={key}
http://localhost:5050/search/contacts?contact={key}

Related Issues 🩹
Resolves #13
Resolves #14

### CODE REVIEW

**Comment by crudemandarin** [comment]
Let's gooo!!!

File: src/services/search-service.ts [hunk 1]
```diff
@@ -1,18 +1,45 @@
...
+		table: string,
+		key: string,
+		columns: string[]
+	): Promise<RowDataPacket[]> {
+		logger.info(
+			`SearchService invoked! Table = ${table}, Key = ${key}, Columns = ${columns}`
+		);
+
+		const whereParams = ` WHERE MATCH (${columns})`;
+		const againstParams = ` AGAINST ('${key}' IN NATURAL LANGUAGE MODE)`;
```
Comment (crudemandarin) [inline src/services/search-service.ts:16]: Does this SQL code here create the index in real-time? Or is there additional SQL code to build the index ahead of API usage?

Comment (crudemandarin) [inline src/services/search-service.ts:16]: Does this SQL code create the index here in real-time? Or is the index created in the database before API usage?

File: src/services/search-service.ts [hunk 2]
```diff
@@ -1,18 +1,45 @@
...
+		]);
+		return rows;
 	}
 
-	static async searchEvents() {
-		logger.info("Search Events");
+	static async searchEvents(key: string): Promise<RowDataPacket[]> {
+		logger.info(`SearchService.searchEvents invoked! Key = ${key}`);
+
+		const rows = SearchService.search("event", key, ["title", "description"]);
```
Comment (crudemandarin) [inline src/services/search-service.ts:41]: What was the result on being able to search DATE columns?

**Review by crudemandarin — COMMENTED** [review]
This is looking great so far. Awesome code style and TS typing. Could you provide clarifications on my comments?

File: src/services/search-service.ts (hunk 1 above)
Comment (crudemandarin) [reply src/services/search-service.ts:16]: If the index is created in the DB beforehand, can you provide the SQL code to create the indices?

Comment (Junnyyy) [reply src/services/search-service.ts:16]: I completely forgot, the SQL code does not create the index it only SELECTS from the index. I left index creation up to us so we can find tune when and how we create the index.

To create the index for **events** view the code snipped below.
```SQL
ALTER TABLE event ADD FULLTEXT(title, description);
```

To create the index for **contacts** view the code snipped below.
```SQL
ALTER TABLE contact ADD FULLTEXT(contact_id, uh_id, email, first_name, last_name, phone_number);
```
! Becareful as the ALTER code can be ran multiple times creating multiple indexes.

If for any reason there is a duplicate creation of an index or you would like to delete an index you can SHOW the table
```SQL
SHOW CREATE TABLE table_name;
```
and delete the `FULLTEXT KEY` index using this code 
```SQL
ALTER TABLE event DROP INDEX fulltext_key_name;
```

File: src/services/search-service.ts (hunk 2 above)
Comment (Junnyyy) [reply src/services/search-service.ts:41]: MySQL fulltext indexes doesn't support `datetime` columns. They only support `CHAR, VARCHAR, TEXT` columns. I implemented the indexes for any section possible aside from Contact table's `shirt_size_id`.

**Comment by crudemandarin** [comment]
Looks great thanks!

**Pull request merged**